### PR TITLE
Add dandelion and chamomile tea bags and recipes

### PR DESCRIPTION
**Pull request by levkropp**
SUMMARY: Content "Add dandelion and chamomile tea bags, resolves #43225"

#### Purpose of change
This PR implements #43225 , which makes the green tea and fruit tea recipes require boiling.

Chamomile and dandelion teas are easily available in bag form (plenty of results on Amazon for example), and should appear wherever other tea bags appear. 

There are now two recipes for dandelion, chamomile, and herbal tea, one requiring cooking skill and raw ingredients, the other requiring no skill and a teabag.

#### Describe the solution

Create dandelion and chamomile tea bags based on the existing implementations of tea bags, then create the recipes for creating the teas with the bags. For multiple recipes with the same result you seem to need an "id_suffix" which I have set to "from_tea_bag" 

These new teabags spawn with the other teabags in the teabag_box item group.

The PR used to change item IDs but that's a big no-no so a consistent naming system for teas will have to wait.

#### Describe alternatives you've considered

Simply merging #43225

Creating even more recipes like a sweetened version for each tea, but someone else can do that if they really like sweetened fruit, herbal, and green teas.

#### Testing

Spawned in as a new character and checked my recipes. Got some water and made one of each tea. Drank the tea and felt good about myself. Once my cooking increased to 1 from boiling tea bags I was able to see the recipe for herbal tea using wild herbs. Looks good!

### CODE REVIEW

**Comment by Fris0uman** [comment]
Please don't change id of item that already exist, it's going to break saves.

**Comment by levkropp** [comment]
> 
> 
> Please don't change id of item that already exist, it's going to break saves.

that explains the origin of the naming inconsistencies, i have reverted all changed item ids back and now the recipes work on new and old saves

**Comment by ZhilkinSerg** [comment]
Tests are failing - calories for new tea bags need to be updated:

```
-------------------------------------------------------------------------------

recipe_permutations

-------------------------------------------------------------------------------

comestible_test.cpp:114

...............................................................................

comestible_test.cpp:151: FAILED:

  CHECK( lower_bound <= mystats.calories.avg() )

with expansion:

  20.8f <= 0.0

RecipeID: dandelion_tea_from_tea_bag, default is 26 Calories,

Current recipe range: 0-0, Average 0

Average recipe Calories must fall within this range, derived from default Calories: 21-31

comestible_test.cpp:152: FAILED:

  CHECK( mystats.calories.avg() <= upper_bound )

with expansion:

  90.0 <= 2.4f

RecipeID: herbal_tea, default is 2 Calories,

Current recipe range: 90-90, Average 90

Average recipe Calories must fall within this range, derived from default Calories: 2-2

comestible_test.cpp:151: FAILED:

  CHECK( lower_bound <= mystats.calories.avg() )

with expansion:

  1.6f <= 0.0

RecipeID: herbal_tea_from_tea_bag, default is 2 Calories,

Current recipe range: 0-0, Average 0

Average recipe Calories must fall within this range, derived from default Calories: 2-2
```

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -1470,6 +1470,39 @@
     "tools": [ [ [ "water_boiling_heat", 2, "LIST" ] ] ],
     "components": [ [ [ "tea_fruit_bag", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
   },
+  {
+    "type": "recipe",
+    "result": "herbal_tea",
+    "id_suffix": "from_tea_bag",
+    "activity_level": "LIGHT_EXERCISE",
+    "copy-from": "tea",
+    "activity_level": "fake",
```
Comment (ZhilkinSerg) [inline data/json/recipes/recipe_food.json:1479]:
```suggestion
```

File: data/json/recipes/recipe_food.json [hunk 2]
```diff
@@ -1470,6 +1470,39 @@
...
+    "tools": [ [ [ "water_boiling_heat", 2, "LIST" ] ] ],
+    "components": [ [ [ "herbal_tea_bag", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
+  },
+  {
+    "type": "recipe",
+    "result": "dandelion_tea",
+    "id_suffix": "from_tea_bag",
+    "activity_level": "LIGHT_EXERCISE",
+    "copy-from": "tea",
+    "activity_level": "fake",
```
Comment (ZhilkinSerg) [inline data/json/recipes/recipe_food.json:1490]:
```suggestion
```

File: data/json/recipes/recipe_food.json [hunk 3]
```diff
@@ -1470,6 +1470,39 @@
...
+    "tools": [ [ [ "water_boiling_heat", 2, "LIST" ] ] ],
+    "components": [ [ [ "tea_bag_dandelion", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
+  },
+  {
+    "type": "recipe",
+    "result": "chamomile_tea",
+    "id_suffix": "from_tea_bag",
+    "activity_level": "LIGHT_EXERCISE",
+    "copy-from": "tea",
+    "activity_level": "fake",
```
Comment (ZhilkinSerg) [inline data/json/recipes/recipe_food.json:1501]:
```suggestion
```

**Comment by mikt4743thesecond** [comment]
Great PR! Aren't these already in a mod called Wild Living? Even if so, glad to see them in mainline now.

**Comment by I-am-Erk** [comment]
The only thing stopping this from merging is that the calories for the teabags need to fit with the tests. Those can be a bit tough to interpret, do you need clarification?

**Comment by levkropp** [comment]
> The only thing stopping this from merging is that the calories for the teabags need to fit with the tests. Those can be a bit tough to interpret, do you need clarification?

I checked the test code and I'm not sure why the tests are failing since tea bags don't have calories (?) And if the original tea bag did that should get copied over to the new tea bags, so I'm kind of confused as to why it's not working

If you know the code to fix it you can just suggest it and I'll implement it 😅

**Comment by I-am-Erk** [comment]
If the bags have no calories, the problem would be that the game expects them to have calories. These tests often confuse me too, to the point that I was considering either removing or totally rewriting them, but I'll see if I can suggest the appropriate fix. I think probably we need to make sure herbal tea and dandelion tea don't expect to have calories

File: data/json/items/comestibles/other.json [hunk 4]
```diff
@@ -825,7 +825,20 @@
     "type": "COMESTIBLE",
     "name": { "str": "herbal tea bag" },
     "copy-from": "tea_bag",
-    "color": "green",
-    "description": "Paper sachet with dried wild herbs inside.  Put it into boiling water to make some healthy warm drink."
+    "description": "A paper sachet with dried wild herbs inside.  Put it into boiling water to make a cup of herbal tea."
```
Comment (I-am-Erk) [inline data/json/items/comestibles/other.json:828]:
```suggestion
    "calories": 27,
    "description": "A paper sachet with dried wild herbs inside.  Put it into boiling water to make a cup of herbal tea."
```

File: data/json/recipes/recipe_food.json [hunk 5]
```diff
@@ -1483,7 +1513,7 @@
     "batch_time_factors": [ 80, 4 ],
     "qualities": [ { "id": "BOIL", "level": 1 } ],
     "tools": [ [ [ "water_boiling_heat", 2, "LIST" ] ] ],
-    "components": [ [ [ "wild_herbs", 10 ], [ "herbal_tea_bag", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
+    "components": [ [ [ "wild_herbs", 10 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
```
Comment (I-am-Erk) [inline data/json/recipes/recipe_food.json:1516]:
```suggestion
    "components": [ [ [ "wild_herbs", 3 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
```
In addition, you will need to change the default calories of herbal tea to 27 in this mode.

**Review by I-am-Erk — COMMENTED** [review]
Honestly I should just remove that test.

File: data/json/items/comestibles/other.json [hunk 6]
```diff
@@ -826,8 +826,23 @@
     "type": "COMESTIBLE",
     "name": { "str": "herbal tea bag" },
     "copy-from": "tea_bag",
-    "color": "green",
-    "description": "Paper sachet with dried wild herbs inside.  Put it into boiling water to make some healthy warm drink."
+    "calories": 27,
```
Comment (kevingranade) [inline data/json/items/comestibles/other.json:829]:
```suggestion
```

File: data/json/items/comestibles/other.json [hunk 7]
```diff
@@ -826,8 +826,22 @@
     "type": "COMESTIBLE",
     "name": { "str": "herbal tea bag" },
     "copy-from": "tea_bag",
-    "color": "green",
-    "description": "Paper sachet with dried wild herbs inside.  Put it into boiling water to make some healthy warm drink."
+    "description": "A paper sachet with dried wild herbs inside.  Put it into boiling water to make a cup of herbal tea."
```
Comment (kevingranade) [inline data/json/items/comestibles/other.json:829]:
```suggestion
    "calories": 2,
    "description": "A paper sachet with dried wild herbs inside.  Put it into boiling water to make a cup of herbal tea."
```

File: data/json/items/comestibles/other.json [hunk 8]
```diff
@@ -826,8 +826,23 @@
     "type": "COMESTIBLE",
     "name": { "str": "herbal tea bag" },
     "copy-from": "tea_bag",
-    "color": "green",
-    "description": "Paper sachet with dried wild herbs inside.  Put it into boiling water to make some healthy warm drink."
+    "calories": 2,
```
Comment (kevingranade) [inline data/json/items/comestibles/other.json:829]:
```suggestion
    "calories": 9,
```

File: data/json/recipes/recipe_food.json [hunk 9]
```diff
@@ -1557,7 +1587,7 @@
     "batch_time_factors": [ 80, 4 ],
     "qualities": [ { "id": "BOIL", "level": 1 } ],
     "tools": [ [ [ "water_boiling_heat", 2, "LIST" ] ] ],
-    "components": [ [ [ "wild_herbs", 10 ], [ "herbal_tea_bag", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
+    "components": [ [ [ "wild_herbs", 3 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
```
Comment (kevingranade) [inline data/json/recipes/recipe_food.json:1590]:
```suggestion
    "components": [ [ [ "wild_herbs", 1 ] ], [ [ "water", 1 ], [ "water_clean", 1 ] ] ]
```

**Pull request merged**